modules/phase_field/src/kernels/ACBarrierFunction.C

```diff
@@ -0,0 +1,104 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ACBarrierFunction.h"
+
+registerMooseObject("PhaseFieldApp", ACBarrierFunction);
+
+template <>
+InputParameters
+validParams<ACBarrierFunction>()
+{
+  InputParameters params = validParams<ACGrGrBase>();
+  params.addRequiredParam<MaterialPropertyName>("gamma", "The gamma value to use with the kernel");
```

Comment:
That doc string is not useful at all! Please spell out the physical significance of gamma.